src/app/about-marketplace/about-marketplace.component.ts

```diff
@@ -40,17 +41,17 @@ export class AboutMarketplaceComponent implements OnInit {
     
     this.delay = AppConstants.CACHE_DELAY;
 
-    this.cacheProducts = localStorage.getItem('cacheProducts');
-
-    this.cacheDate = sessionStorage.getItem('cacheDate');
+    this.cacheDate = localStorage.getItem('cacheDate');
     this.formatDate = this.dp.transform(Date.now(), 'yyyy-MM-dd');
 
-    this.cacheAgencies = sessionStorage.getItem('cacheAgencies');
-    this.cacheAssessors = sessionStorage.getItem('cacheAssessors');
-    this.cacheFilters = sessionStorage.getItem('cacheFilters');
-    this.cacheMetrics = sessionStorage.getItem('cacheMetrics');
-    this.cacheAtoMapping = sessionStorage.getItem('cacheAtoMapping');
-    this.cacheReuseMapping = sessionStorage.getItem('cacheReuseMapping');
+    this.cacheProducts = LZString.decompressFromUTF16(localStorage.getItem('cacheProducts') || '{}');
+    this.cacheAgencies = LZString.decompressFromUTF16(localStorage.getItem('cacheAgencies') || '{}');
+    this.cacheAssessors = LZString.decompressFromUTF16(localStorage.getItem('cacheAssessors') || '{}');
+    this.cacheFilters = LZString.decompressFromUTF16(localStorage.getItem('cacheFilters') || '{}');
+    this.cacheMetrics = LZString.decompressFromUTF16(localStorage.getItem('cacheMetrics') || '{}');
+    this.cacheAtoMapping = LZString.decompressFromUTF16(localStorage.getItem('cacheAtoMapping') || '{}');
+    this.cacheReuseMapping = LZString.decompressFromUTF16(localStorage.getItem('cacheReuseMapping') || '{}');
+
```

Comment:
Why is this operation happening here and also in getJsonData below? This looks like the work is being done twice.

Reply:
Looking further down, this work appears to be done in multiple places. Can this be refactored to use a common shared data object?